### PR TITLE
Adkrawcz/extract values for situations

### PR DESCRIPTION
**Pull request by adamkrawczyk**
#### Description

Fixes #  

#### Where has this been tested?

  * **Platform(s):** ...
  * **Library version:** ...

### CODE REVIEW

File: ad_rss/src/core/RssCheck.cpp [hunk 1]
```diff
@@ -87,22 +87,19 @@ bool RssCheck::calculateProperResponse(world::WorldModel const &worldModel, stat
...
-                               bool dummy)
+logging::ExtendedSituationData &RssCheck::calculateProperResponse(world::WorldModel const &worldModel,
+                                                                  situation::SituationSnapshot &situationSnapshot,
+                                                                  state::RssStateSnapshot &rssStateSnapshot,
+                                                                  state::ProperResponse &properResponse,
+                                                                  bool dummy)
 {
-  // suppress dummy 
+  // suppress dummy
   (void)dummy;
```
Comment (lchojnack) [inline ad_rss/src/core/RssCheck.cpp]: Better to use `static_cast<void>(dummy)`

File: ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp [hunk 2]
```diff
@@ -155,8 +158,10 @@ bool RssStructuredSceneIntersectionChecker::checkIntersectionSafe(Situation cons
                                                           isSafe);
 
       data_intersection.longitudinal_relative_position = "EgoInFront";
```
Comment (lchojnack) [inline ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp:161]: Is it possible to use `to_string()` statement?

File: ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp [hunk 3]
```diff
@@ -120,9 +121,10 @@ bool RssStructuredSceneIntersectionChecker::checkIntersectionSafe(Situation cons
     rssStateInformation.currentDistance = situation.egoVehicleState.distanceToEnterIntersection;
     result = checkStopInFrontIntersection(situation.egoVehicleState, rssStateInformation.safeDistance, isSafe);
 
-    data_intersection.ego_current_distance_to_intersection = static_cast<double>(rssStateInformation.currentDistance);
+    data_intersection.ego_current_distance_to_intersection
+      = static_cast<double>(situation.egoVehicleState.distanceToEnterIntersection);
     data_intersection.ego_safe_distance_to_intersection = static_cast<double>(rssStateInformation.safeDistance);
-    data_intersection.npc_current_distance_to_intersection = -1.0;  // Not calculated
+    data_intersection.npc_current_distance_to_intersection = -1.0; // Not calculated
```
Comment (lchojnack) [inline ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp]: Magic number, could be improved, but not critical

File: ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp [hunk 4]
```diff
@@ -198,7 +199,8 @@ bool RssStructuredSceneNonIntersectionChecker::calculateLongitudinalRssStateSame
                                                         isSafe);
 
     data_non_intersection.longitudinal_relative_position = "EGO in front";
```
Comment (lchojnack) [inline ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp:201]: Is `to_string()` function can be used here?

File: ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp [hunk 5]
```diff
@@ -212,7 +214,8 @@ bool RssStructuredSceneNonIntersectionChecker::calculateLongitudinalRssStateSame
                                                         isSafe);
 
     data_non_intersection.longitudinal_relative_position = "Other in front";
```
Comment (lchojnack) [inline ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp:216]: Is `to_string()` function can be used here?

**Review by lchojnack — APPROVED** [review]
Small changes proposed

File: ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp (hunk 2 above)
Comment (adamkrawczyk) [reply ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp:161]: Could be used however the  `InFront` / `OverlapBack` ... without the "Ego" added is more informative to me

File: ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp (hunk 5 above)
Comment (adamkrawczyk) [reply ad_rss/src/situation/RssStructuredSceneNonIntersectionChecker.cpp:216]: The same as previously -> original text is without ego/other word

**Pull request merged**